Loosen sentence length requirement on Connect/Diagnostic

cissyxyu opened the pull request:
## WHAT
Shorten the minimum sentence length from 2 words shorter than the optimal response, to 4 words shorter than the optimal response.

## WHY
Right now our requirement for sentence minimum length is a bit too strict. It requires sentences to be at most two words shorter than the optimal response, which is too strict.

## HOW
Change the logic in the marking logic to allow for up to 3 words shorter than the optimal response.

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? | YES
Have you deployed to Staging? | Not yet - deploying now!
Self-Review: Have you done an initial self-review of the code below on Github? | Yes
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | Yes

cissyxyu committed:
logic change and package bumps

emilia-friedberg reviewed (commented):
I see the test updates here but I don't actually see a change to the algorithm itself (which makes me wonder if we need negative tests as well as positive ones -- for instance confirming that it returns false if the response string is only three words shorter).

cissyxyu committed:
add more tests and real logic change

cissyxyu commented:
> I see the test updates here but I don't actually see a change to the algorithm itself (which makes me wonder if we need negative tests as well as positive ones -- for instance confirming that it returns false if the response string is only three words shorter).

Added the change. And added new tests to confirm the change!

cissyxyu committed:
add lms package lock
